extensions/lisp-mode/lisp-mode.lisp

```diff
@@ -472,14 +473,19 @@
 (define-command lisp-listen-in-current-package () ()
   (check-connection)
   (alexandria:when-let ((repl-buffer (repl-buffer))
-                        (package (buffer-package (current-buffer))))
+                        (package (buffer-package (current-buffer)))
+                        (original-buffer (current-buffer)))
     (save-excursion
-      (setf (current-buffer) repl-buffer)
-      (destructuring-bind (name prompt-string)
-          (lisp-eval `(micros:set-package ,package))
-        (new-package name prompt-string)))
-    (start-lisp-repl)
-    (buffer-end (buffer-point repl-buffer))))
+      (cond ((find-package package)
```

Comment:
I think it is necessary to check whether the package exists on the server side.
This would search for the package within the client (within the runtime of lem), so it would only behave correctly when lem and the server are in the same runtime.

Reply:
something like this

```
(lisp-eval `(not (null (cl:find-package ,package))))
```

Reply:
New commit made to address this. I keep forgetting there's a server side, my bad.